test/tools/test_tools_time.py

```diff
@@ -47,6 +71,96 @@ def test_validate_format_none():
         time.validate_format(None)
 
 
+def test_get_nick_timezone(db: SopelDB):
+    nick = 'IronMan'
+    assert time.get_nick_timezone(db, nick) is None
+
+    db.set_nick_value(nick, 'timezone', 'Europe/Paris')
+    assert time.get_nick_timezone(db, nick) == 'Europe/Paris'
+
+    db.set_nick_value(nick, 'timezone', 'Invalid_TZ')
+    assert time.get_nick_timezone(db, nick) is None
+
+
+def test_get_channel_timezone(db: SopelDB):
+    channel = '#test'
+    assert time.get_channel_timezone(db, channel) is None
+
+    db.set_channel_value(channel, 'timezone', 'Europe/Paris')
+    assert time.get_channel_timezone(db, channel) == 'Europe/Paris'
+
+    db.set_channel_value(channel, 'timezone', 'Invalid_TZ')
+    assert time.get_channel_timezone(db, channel) is None
+
+
+def test_get_timezone(db: SopelDB, tmpconfig: Config):
+    zone = 'Europe/Paris'
+    zone_nick = 'America/New_York'
+    zone_channel = 'Asia/Tokyo'
+    nick = 'IronMan'
+    channel = '#test'
+
+    assert time.get_timezone(db) is None
+    assert time.get_timezone(db, tmpconfig) == 'UTC', (
+        'Must use default timezone from config'
+    )
+    assert time.get_timezone(db, tmpconfig, zone) == zone
+    assert time.get_timezone(db, tmpconfig, zone, nick, channel) == zone
+
+    # channel tz when no tz and no nick tz (or tz is invalid)
+    db.set_channel_value(channel, 'timezone', zone_channel)
+    assert time.get_timezone(
+        db, tmpconfig, None, nick, channel
+    ) == zone_channel, 'Channel '
+    assert time.get_timezone(
+        db, tmpconfig, 'Invalid/Tz', nick, channel
+    ) == zone_channel
+
+    # nick tz has priority
+    db.set_nick_value(nick, 'timezone', zone_nick)
+    assert time.get_timezone(db, tmpconfig, None, nick, channel) == zone_nick
+
+
+def test_get_timezone_zone_nick_or_channel(db: SopelDB, tmpconfig):
+    zone_nick = 'America/New_York'
+    zone_channel = 'Asia/Tokyo'
+    nick = 'IronMan'
+    channel = '#test'
+
+    db.set_nick_value(nick, 'timezone', zone_nick)
+    db.set_channel_value(channel, 'timezone', zone_channel)
+
+    # when zone is actually a nick
+    assert time.get_timezone(db, tmpconfig, zone=nick) == zone_nick
+    assert time.get_timezone(
+        db, tmpconfig, zone=nick, channel=channel
+    ) == zone_nick, 'zone argument must have priority over channel arguments'
+    assert time.get_timezone(
+        db, tmpconfig, zone=nick, nick=nick, channel=channel
+    ) == zone_nick, 'zone argument must have priority over other arguments'
+
+    # when zone is actually a channel
+    assert time.get_timezone(db, tmpconfig, zone=channel) == zone_channel
+    assert time.get_timezone(
+        db, tmpconfig, zone=channel, nick=nick
+    ) == zone_channel, 'zone argument must have priority over nick argument'
+    assert time.get_timezone(
+        db, tmpconfig, zone=channel, nick=nick, channel=channel
+    ) == zone_channel, 'zone argument must have priority over other arguments'
```

Comment:
@dgw points out that I was reading the doc wrong and that this _is_ documented behavior. I think that probably clinches it, getting rid of this would be too disruptive to plugins without advance notice.

So, I guess my question here is more like _"how do we feel about deprecating that behavior?"_

Reply:
That's what convenient functions do: they are convenient for the dev-user, not the dev-writer. 😁 

I don't mind it, it is, indeed more complicated for us, and at the same time, this is now fully covered by our tests, so it doesn't bother me that much.